File: src/main/java/moe/caa/fabric/quitconfirm/client/modmenu/QuitConfirmModMenu.java
package moe.caa.fabric.quitconfirm.client.modmenu;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import moe.caa.fabric.quitconfirm.client.Screens;
import net.minecraft.client.gui.screen.Screen;

public class QuitConfirmModMenu implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return (ConfigScreenFactory<Screen>) Screens::generateConfigScreen;
    }
}

File: src/main/kotlin/moe/caa/fabric/quitconfirm/client/modmenu/QuitConfirmModMenu.kt
package moe.caa.fabric.quitconfirm.client.modmenu

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import moe.caa.fabric.quitconfirm.client.generateConfigScreen

class QuitConfirmModMenu : ModMenuApi {
    override fun getModConfigScreenFactory() = ConfigScreenFactory {
        return@ConfigScreenFactory generateConfigScreen(it)
    }
}